src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
 {
     return fWalletRbf;
 }
+
+bool WalletModel::nameAvailable(const QString &name)
+{
+    JSONRPCRequest jsonRequest;
+    UniValue params (UniValue::VOBJ);
+    UniValue res, array, isExpired;
+
+    const std::string strName = name.toStdString();
+    params.push_back (Pair("name", strName));
+    jsonRequest.strMethod = "name_show";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = tableRPC.execute(jsonRequest);
+
+    } catch (const UniValue& e) {
+        return true;
```

Comment:
This is dangerous -- you could at least inspect the exception a bit to see if it actually is a "name not found" or some other error.